### ISSUE TITLE
Update DefaultJvmTestSuite and BuildInit to use JUnit 5 when applying useKotlinTest

### ISSUE THREAD

**Issue by Pihanya** [gradle/gradle#19761]
When I try to configure `JvmTestSuite` (`DefaultJvmTestSuite`) with `useKotlinTest("1.6.10")`
Gradle uses JUnit 4 implementation of `kotlin.test` library: [org.jetbrains.kotlin:kotlin-test-junit:1.6.10](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-test-junit/1.6.10). See also: [`DefaultJvmTestSuite.java#L53`](https://github.com/gradle/gradle/blob/v7.3.3/subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java#L53).

When using Kotlin JVM, I expect that `kotlin.test` library with JUnit 5 implementation is included in test compile-time and runtime dependencies: [org.jetbrains.kotlin:kotlin-test-junit5:1.6.10](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-test-junit5/1.6.10).

```build.gradle.kts
plugins {
    kotlin("jvm") version "1.6.10"
}

group = "ru.pihanya.reproduce.gradle"

repositories {
    mavenCentral()
}

dependencies {
    implementation(kotlin("stdlib"))
}

extensions.configure<org.gradle.testing.base.TestingExtension> {
    suites.withType<org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite>()
        .configureEach { useKotlinTest("1.6.10") }
}
```

<img src="https://user-images.githubusercontent.com/20954387/151706051-81a72785-bd80-4676-a3dd-e2110b67b57f.png" width="60%">

### Expected Behavior

[org.jetbrains.kotlin:kotlin-test-junit5](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-test-junit5) is used when call `DefaultJvmTestSuite#userKotlinTest`.

### Current Behavior

[org.jetbrains.kotlin:kotlin-test-junit](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-test-junit) is used when call `DefaultJvmTestSuite#userKotlinTest`.

### Context

Switching to JUnit 5 `kotlin.test` implementation will break backward compatibility. But [Test Suites](https://docs.gradle.org/current/samples/sample_incubating_jvm_multi_project_with_additional_test_types.html) is [incubrating](https://docs.gradle.org/current/userguide/feature_lifecycle.html#sec:incubating_state) feature so there should not be a great problem to break backward compatibility.

**There's also a workaround solution for this problem (see spoiler below).**
<details>
<summary>
<b>Workaround solution</b> (spoiler)
</summary>

```build.gradle.kts
plugins { 
    kotlin("jvm") version "1.6.10"
}

group = "ru.pihanya.reproduce.gradle"

repositories {
    mavenCentral()
}

dependencies {
    implementation(kotlin("stdlib"))
}

extensions.configure<org.gradle.testing.base.TestingExtension> {
    suites.withType<org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite>().configureEach {
        dependencies {
            implementation("org.jetbrains.kotlin:kotlin-test-junit5:1.6.10")
        }
    }
}
```

</details>

### Steps to Reproduce 

1. Clone repository with reproduction of the issue
```shell
git clone git@github.com:Pihanya/gradle-reproduce.git
```
2. Change the current directory to the cloned repository
```shell
cd gradle-reproduce
```
3. Generate Gradle wrapper for the project:
```shell
gradle wrapper --gradle-version 7.3.3
```
4. Reproduce problem by executing task `checkDefaultJvmTestSuiteUsesJunit5`:
```shell
./gradlew checkDefaultJvmTestSuiteUsesJunit5
```

**Final script:**
```shell
git clone git@github.com:Pihanya/gradle-reproduce.git
cd gradle-reproduce
gradle wrapper --gradle-version 7.3.3
./gradlew checkDefaultJvmTestSuiteUsesJunit5
```

### Your Environment

Gradle: `7.3.3`
Build scan URL: https://gradle.com/s/rvizzusvf6gzw

**Comment by Pihanya** [comment]
Gradle 7.4.0 also seems to be released with this bug (see [`DefaultJvmTestSuite.java#L53`](https://github.com/gradle/gradle/blob/16e2cae17bf919033ccc392786eff9f176e295e6/subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java#L53)).

Are there any plans to fix it in Gradle 7.5.0?

I can perform correction by myself and open a pull request with a change of
```java
KOTLIN_TEST("org.jetbrains.kotlin:kotlin-test-junit", "1.6.10")
```
... to
```java
KOTLIN_TEST("org.jetbrains.kotlin:kotlin-test-junit5", "1.6.10")
```
... but I guess there should be some discussion before I do it, am I right?

@tresat, @eskatos should there be some discussion, or it's ok if I create a pull request with the mentioned fix?

**Comment by tresat** [comment]
@Pihanya - This was intentional, to match the template project produced by the build init plugin.  

We don't consider this a bug, but we'd welcome a PR to upgrade this.  You'd need to update `JvmProjectInitDescriptor` and `KotlinGradlePluginProjectInitDescriptor` as well so that the test suites plugin and init plugin both track each other.  Run `org.gradle.testing.testsuites.TestSuitesIntegrationTest` by executing `gradlew :testing-jvm:quickTest sanityCheck` to ensure the tests pass with the new version.
